lib/spack/spack/test/config.py

```diff
@@ -265,6 +265,11 @@ def test_write_to_same_priority_file(mock_low_high_config, compiler_specs):
 
 
 def test_add_config_path(mutable_config):
+    # NOTE:
+    # The config path: "config:install_tree:root:<path>" is unique in that it can accept multiple
+    # schemas (such as a dropped "root" component) which is a typical and may lead to passing tests
+    # when the behavior is in reality incorrect.
```

Comment:
I think it would be more clear to place this comment next to where we set `config:build_stage`, and explain that no sub-key of this accepts a string in our schema (I find it confusing when a comment exclusively refers to the behavior we want to avoid).

(resolving the following is not strictly critical but) Thinking about it more though, my confusion over the original test failing to catch this is renewed:

```
    path = 'config:install_tree:root:"C:/path/to/config.yaml"'
    spack.config.add(path)           <--- this part might add to config:install_tree instead of config:install_tree:root
    set_value = spack.config.get("config")["install_tree"]["root"]       <--- but this part retrieves the key and does not omit "root"
    assert set_value == "C:/path/to/config.yaml"
```